### PR TITLE
lint: add shellcheck and spread-shellcheck linters

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Add new linter for shellcheck and spread-shellcheck to the github workflow.

Includes some minor fixes to be shellcheck-compliant.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -108,6 +108,13 @@ test-pylint:
 test-pyright:
 	pyright .
 
+.PHONY: test-shellcheck
+test-shellcheck:
+	# shellcheck for shell scripts
+	git ls-files | xargs file -N | grep shell.script | cut -f1 -d: | xargs shellcheck
```
Comment (lengau) [inline Makefile:114]: Can you change `starbase` to use this for shellcheck too? It's much better than the current approach IMO

**Review by lengau — APPROVED** [review]

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -14,10 +14,11 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
+          submodules: true
       - name: Install packages
         run: |
           sudo apt update
-          sudo apt install -y libapt-pkg-dev
+          sudo apt install -y libapt-pkg-dev shellcheck
```
[the remark this answers is not in the document]
Comment (mr-cal) [reply .github/workflows/tests.yaml:21]: Nice, I'll do that instead!

File: Makefile (hunk 1 above)
Comment (mr-cal) [reply Makefile:114]: Will do!

**Comment by mr-cal** [comment]
Should I add something in docs about needing the `shellcheck` snap for development?

**Review by tigarmo — APPROVED** [review]
nice! thanks

**Pull request merged**